File: samples/da-snowwizard-cs/SnowWizardApi/Functions/Me.cs
using System.Net;
using Microsoft.Azure.Functions.Worker;
using Microsoft.Azure.Functions.Worker.Http;
using Microsoft.Extensions.Logging;
using SnowWizardApi.Services;

namespace SnowWizardApi.Functions;

public class Me
{
    private readonly ILogger<Me> _logger;
    private readonly SnowProfilesService _profilesService;

    public Me(ILogger<Me> logger, SnowProfilesService profilesService)
    {
        _logger = logger;
        _profilesService = profilesService;
    }

    [Function("me")]
    public async Task<HttpResponseData> Run(
        [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "me/{command?}")] HttpRequestData req,
        string? command)
    {
        // Need to implement authentication to get the address from context,
        // for now lets use Fred Luddy as the current user
        var email = "fred.luddy@example.com";

        try
        {
            _logger.LogInformation("➡️ GET /api/me");
            var profile = await _profilesService.GetProfileAsync(email);
            _logger.LogInformation("   ✅ GET /api/me: {Count} profiles returned", profile.Count);

            var response = req.CreateResponse(HttpStatusCode.OK);
            await response.WriteAsJsonAsync(new { results = profile });
            return response;
        }
        catch (Exception ex)
        {
            _logger.LogError("   ❌ GET /api/me: {Error}", ex.Message);
            var errorResp = req.CreateResponse(HttpStatusCode.InternalServerError);
            await errorResp.WriteAsJsonAsync(new { error = ex.Message });
            return errorResp;
        }
    }
}

File: samples/da-snowwizard-j/SnowWizard/src/functions/me.ts
import { app, HttpRequest, HttpResponseInit, InvocationContext } from "@azure/functions";
import ProfilesApiService from "../services/snow_profiles"

/**
 * This function handles the HTTP request and returns the profile information.
 *
 * @param {HttpRequest} req - The HTTP request.
 * @param {InvocationContext} context - The Azure Functions context object.
 * @returns {Promise<Response>} - A promise that resolves with the HTTP response containing the profile information.
 */

export async function profiles(
    req: HttpRequest,
    context: InvocationContext
  ): Promise<HttpResponseInit> {
    
    // Initialize response.
    const res: HttpResponseInit = {
      status: 200,
      jsonBody: {
        results: [],
      },
    };
  
    try {
        // Need to implement authentication to get the address from context, for now lets use Fred Luddy as the current user
        const email = 'fred.luddy@example.com'
  
        console.log(`➡️ GET /api/me: `);
  
        // Fetch the profile from the ServiceNow API.
        const profile = await ProfilesApiService.getProfile(email);
        res.jsonBody.results = profile ?? [];
        console.log(`   ✅ GET /api/me: response status ${res.status}; ${profile.length} profiles returned`);
        return res;
  
    }
    catch (error) {
      console.error(`   ❌ GET /api/me: ${error}`);
      res.status = 500;
      res.jsonBody = { error: error.message };
      return res;
    }
}

app.http("me", {
    methods: ["GET"],
    authLevel: "anonymous",
    route: "me/{*command}",
    handler: profiles,
  });